examples/util/src/lib.rs

```diff
@@ -1,22 +1,31 @@
 use rand::{thread_rng, Rng};
 use std::process::{Child, Command, Stdio};
 use std::{io::ErrorKind, time::Duration};
+use sysinfo::{System, SystemExt};
 use tokio::time::sleep;
 
 const PROMETHEUS_CONFIG_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/prometheus.yml");
 
-pub struct ChildGuard(Child);
+pub struct ChildGuard(Option<Child>);
```

Comment:
I think it would make more sense to have the method return an `Option<ChildGuard>`, rather than having the option inside this struct

Reply:
Actually now looking into it it was done rather of stupidity,  it even less code and more elegant.